tensorflow/compiler/xla/service/dump.cc

```diff
@@ -381,7 +381,8 @@ static bool IsTrivial(const HloComputation& computation) {
 static std::vector<std::string> DumpHloModuleImpl(
     const HloModule& module, const BufferAssignment* buffer_assn,
     const HloExecutionProfile* profile, string_view prefix, string_view suffix,
-    const CanonicalDebugOptions& opts) {
+    const CanonicalDebugOptions& opts,
+    bool dump_top_level = false) {
```

Comment:
This option is confusing, especially since it's only used in one of the branches. Maybe remove it altogether? Are the re callers who want to keep passing `false`?

Reply:
I can change the name or document it better for sure.
I added this as I do not want 2 dump of the before_optimizations.txt file.
Is documentation of that it does good or you want something else?

Reply:
Why would it dump before_optimizations.txt twice? Isn't dumping twice logic located only around HTML?

Reply:
The call to dump before optimizations:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/service.cc#L356
`DumpHloModuleIfEnabled(*module, kBeforeOptimizationsDumpName);`

The call to after optimizations:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/gpu_compiler.cc#L1045-L1047
```
  DumpHloModuleIfEnabled(*hlo_module, *results->buffer_assignment,
                         absl::StrCat("sm_", cuda_compute_capability.ToString(),
                                      "_gpu_after_optimizations"));
```

DumpHloModuleIfEnabled is called by both function. Both will end up calling DumpHloModuleImpl() and that function will dump all the formats.
As it is shared, if I do not make a distinction in both call, both will be dumped.

So I need that new parameter in DumpHloModuleIfEnabled().
In DumpHloModuleImpl(), it could be moved in `const CanonicalDebugOptions& opts` if you prefer. The change will take a little bit more line of code, but would be cleaner I guess.

Reply:
I suppose you talk about the dump_top_level parameter here, not dump_as_html. This PR doesn't add a new XLA flag anymore.

At a high level, now we dump 3 files:
...before_optimizations.txt
...after_optimizations.txt
...after_optimizations.top_level.txt

It is useless to dump a file ...before_optimizations.top_level.txt. dump_top_level is there to prevent that.

Reply:
Oh, do you want to dump two .txt files as well? I thought it was only for HTML. For .txt, I think it's excessive to have two by default.

Reply:
@nouiz Ping ^

Reply:
Sorry, I mean that this PR want to generates:
At a high level, now we dump 3 files:
...before_optimizations.html
...after_optimizations.html
...after_optimizations.top_level.html
 (not .txt)
It is useless to dump a file ...before_optimizations.top_level.txt and dump_top_level is there to prevent that.

I tested manually this PR again and something didn't work as expected. I'll double check if it is my local test that was wrong of if it is the PR.

Too much meeting this week, so this doesn't advance fast.

Reply:
This PR looks good to me. I did a manipulation error when testing it.
What it does it to dump a new file: ...after_optimizations.top_level.html
It only dump a new html file when --xla_dump_hlo_as_html is used.

I edited my comment that had bad file extension.